engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -908,7 +908,7 @@ private void execute() throws YarnException, IOException
           LOG.info("Releasing {} as resource with priority {} was already assigned", allocatedContainer.getId(), allocatedContainer.getPriority());
           releasedContainers.add(allocatedContainer.getId());
           numReleasedContainers++;
-          numRequestedContainers--;
+          numRequestedContainers++;    // undo the decrement above for this allocated container
```

Comment:
@vrozov  Prior change was tested by sending the private patch to the user who reported the problem and he verified it. This change has not been tested. There are no unit tests and this particular condition is difficult/impossible to simulate with actual Hadoop.

Regarding line 925: you could be right but I don't know for sure. What I would suggest (like what I did before) is we should refactor this code as part of a separate JIRA. The refactoring would consist of splitting this code into multiple smaller functions (or even classes) and use mocks to unit test these smaller chunks. For example we can create a class and move all the requested/released container list and count management in this class. This way the code in StreamingAppMasterService.execute() will be manageably small.